src/rust/lib/graph-editor/src/component/visualization.rs

```diff
@@ -0,0 +1,332 @@
+//! This module defines the visualization widgets and related functionality.
+//!
+//! At the core of this functionality is the `Visualisation` that takes in data and renders an
+//! output visualisation which is displayed in a `Container`. The `Container` provides generic UI
+//! elements that facilitate generic interactions, for example, visualisation selection. The
+//! `Container` also provides the FRP API that allows internal interaction with the
+//! `Visualisation`. Data for a visualisation has to be provided wrapped in the `Data` struct.
+//!
+use crate::prelude::*;
+
+use crate::frp;
+
+use ensogl::display::DomSymbol;
+use ensogl::display::object::class::Object;
+use ensogl::display::object::class::ObjectOps;
+use ensogl::display;
+use ensogl::system::web;
+use web::StyleSetter;
+
+
+// ============================================
+// === Wrapper for Visualisation Input Data ===
+// ============================================
+
+/// Wrapper for data that can be consumed by a visualisation.
+/// TODO replace with better typed data wrapper.
+#[derive(Clone,Debug)]
+#[allow(missing_docs)]
+pub enum Data {
+    JSON { content : String },
+    Empty,
+}
+
+impl Data {
+    /// Render the data as JSON.
+    pub fn as_json(&self) -> String {
+        match &self {
+            Data::JSON { content } => content.clone(),
+            Data::Empty => { "{}".to_string() },
+        }
+    }
+}
+
+impl Default for Data{
+    fn default() -> Self {
+        Data::Empty
+    }
+}
+
+
+
+// =============================================
+// === Internal Visualisation Representation ===
+// =============================================
+
+/// Inner representation of a visualisation.
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct Visualization {
+    content : Rc<DomSymbol>
+}
+
+impl Object for Visualization {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.content.display_object()
+    }
+}
+
+impl Visualization {
+    /// Update the visualisation with the given data.
+    // TODO remove dummy functionality and use an actual visualisation
+    pub fn set_data(&self, data:Data){
+                self.content.dom().set_inner_html(
+                    &format!(r#"
+<svg>
+  <circle style="fill: #69b3a2" stroke="black" cx=50 cy=50 r={}></circle>
+</svg>
+"#, data.as_json()));
+     }
+
+    /// Make the visualisation visible in the scene.
+    pub fn show(&self) {
+        // TODO investigate how to do this through display_object functionality.
+        self.content.dom().set_style_or_panic("visibility", "visible");
```

Comment:
FYI - all html elements (and webgl elements too) should appear on the scene as soon as you do something like `scene.add_child(&element)`. They should be hidden as soon as you do `element.unset_parent()`. See `display::ObjectOps` for more info. In case it would not work (it should, just in case as we were recently changing a lot of functionality around it), please ping me! :)